client/tasks.py

```diff
@@ -0,0 +1,227 @@
+"""
+Celery tasks to handle api requests of write api of NodeBB.
+"""
+from logging import getLogger
+
+from celery.task import task
+from django.conf import settings
+from django.conf import settings as django_settings
+from openedx.features.openedx_nodebb_discussion.client.categories import NodeBBCategory
+from openedx.features.openedx_nodebb_discussion.client.groups import NodeBBGroup
+from openedx.features.openedx_nodebb_discussion.client.users import NodeBBUser
+from openedx.features.openedx_nodebb_discussion.client.utils import (
+    get_category_id_from_course_id, get_group_slug_from_course_id,
+    get_group_slug_from_category_id, get_group_name_from_course_id,
+    get_nodebb_uid_from_username
+)
+
+RETRY_DELAY = django_settings.NODEBB_SETTINGS['TASK_RETRY_DELAY']
+
+log = getLogger(__name__)
+
+
+def handle_response(caller, task_name, job_type, status_code, response, entity):
+    """
+    Logs the response of the specific NodeBB API call and retry the task if it fails due to the server inactivity.
+
+    Args:
+        caller (method): Method from which response is logged.
+        task_name (str): Task Name for which we are logging response
+        job_type (str): Job Type for which we are logging response
+        status_code (int): Api response code through which we will check our status.
+        response (str): Received response from Api which we will log.
+        entity (str): Can be anything like course_name or username or group_name
+    """
+    if status_code >= 500:
+        """
+        In case of any internal server error, retry that task again.
+        """
+        log.warning('Retrying: {} task for {}: {}'.format(task_name, job_type, entity))
+        caller.retry()
+    elif status_code >= 400:
+        """
+        In case of any unauthorized request we don't need to retry the task so we log error.
+        """
+        log.error('Failure: {} task for {}: {}, status_code: {}, response: {}'
+                  .format(task_name, job_type, entity, status_code, response))
+    elif 200 <= status_code < 300:
+        """
+        In case of success, just log the success status.
+        """
```

Comment:
These comments are unnecessary.

Reply:
In case of success, the comment might be unnecessary. But in the case of unauthorized and server error comment is necessary because everyone doesn't know about codes and he/she will not be able to understand why we are not retrying there or why we are retrying.